### PR TITLE
feat: add screen request in flutter

### PR DESCRIPTION
**Pull request by uziasferreirazup**
### Related Issues

#1309

### Description and Example

adding the possibility for the user to be able to make a request, however this needs to be discussed in a daily because the web core is not prepared for this type of request

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/3c001c0474e50e9c787fc80386216ac2054839d1/sample/lib/main.dart#L21">sample/lib/main.dart#L21</a> - Unused import: 'package:beagle/networking/beagle_http_method.dart'</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_flutter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: flutter/beagle/lib/beagle_widget.dart [hunk 1]
```diff
@@ -31,17 +32,21 @@ typedef OnCreateViewListener = void Function(BeagleView view);
...
-  /// Server URL will be used to make a request.
-  final String url;
-
   /// that represents a local screen to be shown.
   final String screenJson;
 
+  /// that represents a controllerId.
+  final String controllerId;
+
+  /// that represents a screen request.
```
Comment (matheusribeirozup) [inline flutter/beagle/lib/beagle_widget.dart:47]:
```suggestion
  /// provides the url, method, headers and body to the request.
```

File: flutter/beagle/lib/beagle_widget.dart [hunk 2]
```diff
@@ -31,17 +32,21 @@ typedef OnCreateViewListener = void Function(BeagleView view);
...
+    this.screenRequest,
   }) : super(key: key);
 
-  /// Server URL will be used to make a request.
-  final String url;
-
   /// that represents a local screen to be shown.
   final String screenJson;
 
+  /// that represents a controllerId.
```
Comment (matheusribeirozup) [inline flutter/beagle/lib/beagle_widget.dart:44]: What does the `controllerId` do?

File: flutter/beagle/lib/networking/beagle_screen_request.dart [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'package:beagle/networking/beagle_http_method.dart';
+import 'package:beagle/networking/beagle_network_options.dart';
+
+/// BeagleRequest is used to do requests.
+class BeagleScreenRequest extends BeagleNetworkOptions {
```
Comment (paulomeurerzup) [inline flutter/beagle/lib/networking/beagle_screen_request.dart:21]: What about using composition over inheritance here? This way, we can avoid undesired behavior changes in BeagleScreenRequest due to changes in BeagleNetworkOptions

File: flutter/beagle/lib/beagle_widget.dart (hunk 2 above)
Comment (uziasferreirazup) [reply flutter/beagle/lib/beagle_widget.dart]: I removed the attribute because it is useless for now

**Review by paulomeurerzup — APPROVED** [review]

**Review by matheusribeirozup — APPROVED** [review]

**Pull request merged**